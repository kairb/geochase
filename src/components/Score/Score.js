import React from 'react';
import { makeStyles } from '@material-ui/core';
import PropTypes from 'prop-types';

const useStyles = makeStyles({
  score_inner:{
    borderStyle: 'inset',
    borderWidth: '2px',
    borderColor: 'grey',
    backgroundColor: 'red',
    borderRadius: '5px',
    padding:'5px',
    minWidth: '50px',
    display: 'flex',
    justifyContent: 'space-evenly',
    alignItems: 'centre',
    color:'white'
  },
  score_outer:{
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderStyle: 'solid',
    borderWidth: '1px',
    borderColor: 'grey',
    backgroundColor: 'lightGray',
    borderRadius: '5px',
    padding: '5px',
    minWidth: '110px'
  }
});

const Score = ({score}) => {
  const classes = useStyles();
  return(
    <div className ={classes.score_outer}>
        Score
      <div className = {classes.score_inner}>
          {score}
      </div>
    </div>
  );
}

Score.prototype = {
  score: PropTypes.number.isRequired
}

export default Score;