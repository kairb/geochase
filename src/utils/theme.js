import {createMuiTheme} from '@material-ui/core';

export default createMuiTheme(() => ({
  palette:{
    primary:'green',
    secondary:'white',
  }
}))