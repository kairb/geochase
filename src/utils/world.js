import ground1 from '../assets/scenery/square-1.png';
import ground2 from '../assets/scenery/square-2.png';
import ground3 from '../assets/scenery/square-3.png';
import ground4 from '../assets/scenery/square-4.png';
import ground5 from '../assets/scenery/square-dark.png';

const getRandomGround = () => {
  return `ground${parseInt(Math.random() * 4 + 1)}`;
};

export const createWorld = () => {
  // world goes thisa way
  //  |
  //  |
  //  V
  const getInitialRow = () => {
    return [0, 0, 0, 0, 0, 0, 0, 0, 0, 'ground1'];
  };
  return new Array(40).fill(getInitialRow(), 0, 40);
};

export const blockMap = {
  0: {
    // color: 'none',
  },
  1: {
    // color: 'green',
  },
  ground1: { image: ground1 },
  ground2: { image: ground2 },
  ground3: { image: ground3 },
  ground4: { image: ground4 },
  ground5: { image: ground5 },
};

const getRandom = () => {
  return Math.random() > 0.5 ? "ground5" : 0;
};

export const getCollisionHeight = (world) => {
  const height1 = getColumnWorldHeight(world[18]);
  const height2 = getColumnWorldHeight(world[19]);
  const height3 = getColumnWorldHeight(world[20]);
  const height4 = getColumnWorldHeight(world[21]);
  return [height1, height2, height3, height4].sort()[0];
};

const getColumnWorldHeight = (column) => {
  return ((column.join('').indexOf('ground') - 1) / 10) * window.innerHeight;
};

const getNextBlock = (above, below, index) => {
  if (above !== 0) {
    return 'ground5';
  } else if (below === 0) {
    return 0;
  } else {
    return getRandom();
  }
};

const getNextColumn = (column) => {
  return column
    .map((block, index) => {
      if (index < 2) {
        return 0;
      } else if (index === 9) {
        return 'ground5';
      } else {
        return getNextBlock(column[index - 1], column[index + 1], index);
      }
    })
    .map((block, index, newWorld) => {
      console.log(newWorld);
      if (!newWorld[index - 1] && newWorld[index] && !newWorld[index-2]) {
        return getRandomGround();
      }
      if (newWorld[index - 1] && newWorld[index + 1]) {
        return 'ground5';
      }
      return block;
    });
};

export const updateWorld = (world) => {
  return [...world.slice().splice(1, 39), getNextColumn(world[39])];
};
