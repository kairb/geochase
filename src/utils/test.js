import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider} from '@material-ui/core/styles';
import theme from './theme';
export const renderWithThemeWrapper = (component) =>
  render(
    <ThemeProvider theme={theme}>
      {component}
    </ThemeProvider>
  );