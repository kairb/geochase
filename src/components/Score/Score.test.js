import React from 'react';
import { renderWithThemeWrapper } from '../../utils/test';
import Score from './Score';

describe('Score component', () => {
  it('should render with score', () => {
    const { container, getByText } = renderWithThemeWrapper(
      <Score score={1} />,
    );
    expect(getByText('1')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
