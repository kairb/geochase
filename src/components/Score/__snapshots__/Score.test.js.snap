// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score component should render with score 1`] = `
<div
  class="makeStyles-score_outer-2"
>
  Score
  <div
    class="makeStyles-score_inner-1"
  >
    1
  </div>
</div>
`;
